src/V3MergeCond.cpp

```diff
@@ -51,71 +79,363 @@
+                conditionp = conditionalp->condp();
+            }
+        } else if (const AstNodeIf* const ifp = VN_CAST(nodep, NodeIf)) {
+            conditionp = ifp->condp();
         }
+        while (AstCCast* const castp = VN_CAST(conditionp, CCast)) conditionp = castp->lhsp();
+        return conditionp;
+    }
+
+    void analyzeStmt(AstNodeStmt* nodep, bool tryCondMatch) {
+        VL_RESTORER(m_propsp);
+        // Keep hold of props of enclosing statement
+        StmtProperties* const outerPropsp = m_propsp;
+        // Grab the props of this statement
+        m_propsp = &m_stmtProperties(nodep);
+
+        // Extract condition from statement
+        if (AstNode* const condp = extractCondition(nodep)) {
+            // Remember condition node. We always need this as it is used in the later
+            // traversal.
+            m_propsp->m_condp = condp;
+            // If this is a conditional statement, try to find an earlier one with the same
+            // condition in the same list (unless we have been told not to bother because we know
+            // this node is in a singleton list).
+            if (tryCondMatch) {
+                // Grab the duplicate finder of this list
+                V3DupFinder& dupFinder = m_stack.back();
+                // Find a duplicate condition
+                const V3DupFinder::iterator& dit = dupFinder.findDuplicate(condp);
+                if (dit == dupFinder.end()) {
+                    // First time seeing this condition in the current list
+                    dupFinder.insert(condp);
+                    // Remember last statement with this condition (which is this statement)
+                    condp->user5p(nodep);
+                } else {
+                    // Seen a conditional with the same condition earlier in the current list
+                    AstNode* const firstp = dit->second;
+                    // Add to properties for easy retrieval during optimization
+                    m_propsp->m_prevWithSameCondp = static_cast<AstNodeStmt*>(firstp->user5p());
+                    // Remember last statement with this condition (which is this statement)
+                    firstp->user5p(nodep);
+                }
+            }
+        }
+
+        // Analyse this statement
+        analyzeNode(nodep);
+
+        // If there is an enclosing statement, propagate properties upwards
+        if (outerPropsp) {
+            // Add all rd/wr vars to outer statement
+            outerPropsp->m_rdVars.insert(m_propsp->m_rdVars.cbegin(), m_propsp->m_rdVars.cend());
+            outerPropsp->m_wrVars.insert(m_propsp->m_wrVars.cbegin(), m_propsp->m_wrVars.cend());
+            // If this statement is impure, the enclosing statement is also impure
+            if (m_propsp->m_isFence) outerPropsp->m_isFence = true;
+        }
+    }
+
+    void analyzeVarRef(AstVarRef* nodep) {
+        const VAccess access = nodep->access();
```

Comment:
```suggestion
        const VAccess& access = nodep->access();
```

Reply:
by-value on purpose as we do not try to modify it, and is smaller than a pointer.